pkg/build/api/v1/conversion.go

```diff
@@ -91,6 +91,13 @@ func Convert_v1_BuildTriggerPolicy_To_api_BuildTriggerPolicy(in *BuildTriggerPol
 	switch in.Type {
 	case ImageChangeBuildTriggerTypeDeprecated:
 		out.Type = newer.ImageChangeBuildTriggerType
+		if in.ImageChange == nil && out.ImageChange == nil {
+			out.ImageChange = &newer.ImageChangeTrigger{}
+		}
+	case ImageChangeBuildTriggerType:
+		if in.ImageChange == nil && out.ImageChange == nil {
+			out.ImageChange = &newer.ImageChangeTrigger{}
+		}
```

Comment:
I believe I've at least partially sorted it out ... about to push an update after some more testing, but I'll post a preliminary answer now:

1) I believe the TODO comment was in reference to this section of code: https://github.com/openshift/origin/blob/master/pkg/build/api/v1/conversion_generated.go#L319-L323  ... with no pointer set, we don't enter this for loop, with the call to autoConvert* and its call to DefaultingInterface

2) Then, in this method, we were not calling the autoConvert* method with its call to DefaultingInterfaces, which lead to the call to the Defaulter

3) Tracing through the DefaulConvert call, it does not appear to access the defaultingFunction/defaultingInterfaces list

So I'm removing the above change, and switching from DefaultConvert to autoConver* for the BuildTriggerPolicy .... doing some more testing but it appears to work, and I see using of autoConvert_v1_to_api for other types in this file, so I don't think I'm running afoul of any patterns.